### PR TITLE
adding-glacier-exchange

### PR DESCRIPTION
**Pull request by discochuck**
Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

File: models/glacier_exchange/avalanche_c/glacier_exchange_avalanche_c_schema.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            - blockchain
+            - project
+            - version
+            - tx_hash
+            - evt_index
+            - trace_address
+      - check_dex_seed:
+          blockchain: avalanche_c 
+          project: glacier_exchange
+          version: 1
```
Comment (Hosuke) [inline models/glacier_exchange/avalanche_c/glacier_exchange_avalanche_c_schema.yml]: Hi @discochuck,
in order for `check_dex_seed` to work, can you add some seeds in this file?
https://github.com/duneanalytics/spellbook/blob/main/seeds/dex/trades/dex_trades_seed.csv

File: models/glacier/avalanche_c/glacier_avalanche_c_schema.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            - blockchain
+            - project
+            - version
+            - tx_hash
+            - evt_index
+            - trace_address
+      - check_dex_seed:
+          blockchain: avalanche_c 
+          project: glacier
+          version: 1
```
Comment (Hosuke) [inline models/glacier/avalanche_c/glacier_avalanche_c_schema.yml:27]: Hi @discochuck 
in order for check_dex_seed to work, we may need to add some seeds into the following file:
https://github.com/duneanalytics/spellbook/blob/main/seeds/dex/trades/dex_trades_seed.csv

You can follow the pattern and fill in one or two seed records for testing.

**Review by Hosuke — CHANGES_REQUESTED** [review]

File: models/glacier/avalanche_c/glacier_avalanche_c_schema.yml (hunk 2 above)
Comment (discochuck) [reply models/glacier/avalanche_c/glacier_avalanche_c_schema.yml:27]: i _think_ that should do it.

File: seeds/dex/trades/dex_trades_seed.csv [hunk 3]
```diff
@@ -340,4 +340,5 @@ arbitrum,kyberswap,elastic,2023-03-26,0x070e704ea6da35b05c97e2a2f9ccf01136f9bdf6
...
 bnb,sushiswap,1,2021-11-01 0:00,0xffc3e47949b600ee61d7b3b7b57da6137ed4c55652da396d02ff3ece60bfdd9b,591,0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c,0.01789196956,0x049d68029688eabf473097a2fc38ef61633a3c7a,9.541071
 polygon,sushiswap,1,2021-06-16 0:00,0xc625dc2c5e923afd8469e9e03233d37eab64d627a4e6f221c8b63efcf276523c,179,0x7ceb23fd6bc0add59e62ac25578270cff1b9f619,0.1131447356,0x0d500b1d8e8ef31e21c99d1db9a6444d3adf1270,191.734491
-polygon,sushiswap,1,2021-06-16 0:00,0xa727f3165a6e7533cbeecfbe67e4072f8b9a0eec486a4c9976b25b902a1bd6f0,310,0x7ceb23fd6bc0add59e62ac25578270cff1b9f619,0.0008197673253,0x0d500b1d8e8ef31e21c99d1db9a6444d3adf1270,1.388941988
\ No newline at end of file
+polygon,sushiswap,1,2021-06-16 0:00,0xa727f3165a6e7533cbeecfbe67e4072f8b9a0eec486a4c9976b25b902a1bd6f0,310,0x7ceb23fd6bc0add59e62ac25578270cff1b9f619,0.0008197673253,0x0d500b1d8e8ef31e21c99d1db9a6444d3adf1270,1.388941988
+avalanche_c,glacier,1,2023-04-21,0x8e5b51477a54c3de4e5850c09f33958ca357baa676bb7342389030bc4ba3e901,5,0x3712871408a829c5cd4e86da1f4ce727efcd28f6,60.778340365847253021,0xb31f66aa3c1e785363f0875a1b74e27b85fd66c7,0.1
```
Comment (Hosuke) [inline seeds/dex/trades/dex_trades_seed.csv:344]: Also from the explorer, the token_bought_amount may be `63.768883740248004859`, but I am not very sure, please verify this part.

https://snowtrace.io/tx/0x8e5b51477a54c3de4e5850c09f33958ca357baa676bb7342389030bc4ba3e901

```suggestion
avalanche_c,glacier,1,2023-04-21,0x8e5b51477a54c3de4e5850c09f33958ca357baa676bb7342389030bc4ba3e901,5,0x3712871408a829c5cd4e86da1f4ce727efcd28f6,63.768883740248004859,0xb31f66aa3c1e785363f0875a1b74e27b85fd66c7,0.1
```

Comment (Hosuke) [reply seeds/dex/trades/dex_trades_seed.csv:344]: https://dune.com/queries/2399354?d=1
And the block_date for this trade should be 2023-04-22 in UTC time.

Comment (discochuck) [reply seeds/dex/trades/dex_trades_seed.csv]: doh, thx

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @discochuck 

- Metadata & naming conventions ✅
- Config blocks match uniqueness tests in schema files ✅
- Tests and seeds added for the new models ✅
- Incremental filters added when joining large tables ✅
- Contract addresses are lowercase ✅
- Inner join to large tables(avlanche_c.transactions) as possible ✅
- `CAST()` used in the base models ✅

[Sample glacier_avalanche_c_trades](https://dune.com/queries/2400477?d=1)

**Comment by discochuck** [comment]
I think that's it

**Pull request merged**